src/sentry/migrations/0563_add_slug_reservation_model.py

```diff
@@ -0,0 +1,80 @@
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0562_drop_xactor_actor_from_state"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="OrganizationSlugReservationReplica",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("slug", models.SlugField(unique=True)),
+                (
+                    "organization_id",
+                    sentry.db.models.fields.bounded.BoundedBigIntegerField(db_index=True),
+                ),
+                ("region_name", models.CharField(max_length=48)),
+                ("user_id", sentry.db.models.fields.bounded.BoundedBigIntegerField(db_index=True)),
+                ("date_added", models.DateTimeField(default=django.utils.timezone.now)),
+            ],
+            options={
+                "db_table": "sentry_organizationslugreservationreplica",
+            },
+        ),
+        migrations.CreateModel(
+            name="OrganizationSlugReservation",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("slug", models.SlugField(unique=True)),
+                (
+                    "organization_id",
+                    sentry.db.models.fields.bounded.BoundedBigIntegerField(db_index=True),
+                ),
+                ("user_id", sentry.db.models.fields.bounded.BoundedBigIntegerField(db_index=True)),
+                ("region_name", models.CharField(max_length=48)),
+                (
+                    "reservation_type",
+                    sentry.db.models.fields.bounded.BoundedBigIntegerField(default=0),
+                ),
+                (
+                    "date_added",
+                    models.DateTimeField(default=django.utils.timezone.now, editable=False),
+                ),
+            ],
+            options={
+                "db_table": "sentry_organizationslugreservation",
+                "unique_together": {("organization_id", "reservation_type")},
+            },
```

Comment:
Do we need unique constraints on `slug`? If two orgs have the `acme` slug, which organization actually has the slug?

Reply:
Yep! I've already added the unique constraint to slug which should be on line 59 of the migration:
```("slug", models.SlugField(unique=True)),```

Reply:
Sorry, I missed that 🤦